### PR TITLE
Refactor FirestoreConnector to leverage the firestore-beam-connector from Apache Beam v2.34.0

### PR DESCRIPTION
**Pull request by BenWhitehead**

### CODE REVIEW

File: src/test/java/com/google/exposurenotification/privateanalytics/ingestion/pipeline/IngestionPipelineIT.java [hunk 1]
```diff
@@ -352,7 +352,11 @@ public void testFirestoreReaderPartitionsQueryAndReadsCorrectNumberDocuments()
     seedDatabaseAndReturnEntryVal(numDocs);
 
     PCollection<Long> numShares =
-        testPipeline.apply(new FirestoreReader(CREATION_TIME)).apply(Count.globally());
+        testPipeline
+            .apply(new FirestorePartitionQueryCreation(CREATION_TIME))
+            .apply(FirestoreIO.v1().read().partitionQuery().build())
+            .apply(FirestoreIO.v1().read().runQuery().build())
+            .apply(Count.globally());
```
Comment (gurayAlsac) [inline src/test/java/com/google/exposurenotification/privateanalytics/ingestion/pipeline/IngestionPipelineIT.java]: Further down below we do some assertions on some metrics in the now-deleted FirestoreConnector:
https://github.com/google/exposure-notifications-private-analytics-ingestion/blob/a57faa7cdfafc5688744ca50fc439b79b64947d0/src/test/java/com/google/exposurenotification/privateanalytics/ingestion/pipeline/IngestionPipelineIT.java#L373

https://github.com/google/exposure-notifications-private-analytics-ingestion/blob/a57faa7cdfafc5688744ca50fc439b79b64947d0/src/main/java/com/google/exposurenotification/privateanalytics/ingestion/pipeline/FirestoreConnector.java#L101

Are there equivalents in the new FirestoreIO?

**Comment by hostirosti** [comment]
I think the first step is to go back to a non-snapshot version for the beam SDK. We don't have support for SNAPSHOT versions (we don't include the SNAPSHOT maven repos in our builds)

```
Error:  Failed to execute goal on project enpa-ingestion-model: Could not resolve dependencies for project com.google.exposurenotification.privateanalytics.ingestion:enpa-ingestion-model:jar:20210413_RC01-SNAPSHOT: Could not find artifact org.apache.beam:beam-sdks-java-core:jar:2.32.0-SNAPSHOT -> [Help 1]
Error:  
Error:  To see the full stack trace of the errors, re-run Maven with the -e switch.
Error:  Re-run Maven using the -X switch to enable full debug logging.
Error:  
Error:  For more information about the errors and possible solutions, please read the following articles:
Error:  [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/DependencyResolutionException
Error:  
Error:  After correcting the problems, you can resume the build with the command
Error:    mvn <args> -rf :enpa-ingestion-model
exiting with 1
Error: Process completed with exit code 1.
```

**Comment by hostirosti** [comment]
Looks like `org.apache.beam.sdk.io.gcp.firestore` doesn't exist in version `2.31.0` thus we will need to wait till `2.32.0` is released  .

```
Error:  /home/runner/work/exposure-notifications-private-analytics-ingestion/exposure-notifications-private-analytics-ingestion/src/main/java/com/google/exposurenotification/privateanalytics/ingestion/pipeline/FirestoreConnector.java:[34,44] package org.apache.beam.sdk.io.gcp.firestore does not exist
Error:  /home/runner/work/exposure-notifications-private-analytics-ingestion/exposure-notifications-private-analytics-ingestion/src/main/java/com/google/exposurenotification/privateanalytics/ingestion/pipeline/DeletionPipeline.java:[25,44] package org.apache.beam.sdk.io.gcp.firestore does not exist
```

File: src/test/java/com/google/exposurenotification/privateanalytics/ingestion/pipeline/IngestionPipelineIT.java (hunk 1 above)
Comment (BenWhitehead) [reply src/test/java/com/google/exposurenotification/privateanalytics/ingestion/pipeline/IngestionPipelineIT.java]: There is not a specific counter for partition count, as the PartitionQuery handler is scoped larger than a single PartitionQueryRequest. The number of Partitions resolved would map to the number of output RunQueryRequests + 1 from the PCollection. 

Is the number of partitions important, or are we just trying to ensure that it is doing something useful?

If we're just wanting to make sure a partition query rpc was issued, there is a metric we could look at for that.

Comment (gurayAlsac) [reply src/test/java/com/google/exposurenotification/privateanalytics/ingestion/pipeline/IngestionPipelineIT.java]: We were mostly trying to ensure it was something sensical, being our own custom implementation. Switching to The Official Connector Implementation ™ that's probably no longer a concern and we can remove the counter assertions.

Comment (BenWhitehead) [reply src/test/java/com/google/exposurenotification/privateanalytics/ingestion/pipeline/IngestionPipelineIT.java]: I figured out a way that we can PAssert on the number of queries returned from partition query, so we can ensure there is more than 1 partition.

Comment (gurayAlsac) [reply src/test/java/com/google/exposurenotification/privateanalytics/ingestion/pipeline/IngestionPipelineIT.java]: LGTM, thanks! Resolving.

**Review by gurayAlsac — DISMISSED** [review]

**Comment by hostirosti** [comment]
Waiting for Beam SDK 2.32.0 to be released.

**Comment by BenWhitehead** [comment]
Blocked until https://github.com/apache/beam/pull/15425 is released.

Several collections can have small sizes where a PartitionQueryRequest returns zero cursors and falls into the case of needing the above fix.

**Comment by hostirosti** [comment]
@BenWhitehead apologies for the delay. 
I still see the following integration test failures / errors:

```
[ERROR] Tests run: 4, Failures: 2, Errors: 1, Skipped: 0, Time elapsed: 180.174 s <<< FAILURE! - in com.google.exposurenotification.privateanalytics.ingestion.pipeline.IngestionPipelineIT
[ERROR] com.google.exposurenotification.privateanalytics.ingestion.pipeline.IngestionPipelineIT.testIngestionPipelineAWS  Time elapsed: 5.349 s  <<< ERROR!
com.amazonaws.services.s3.model.AmazonS3Exception: The XML you provided was not well-formed or did not validate against our published schema (Service: Amazon S3; Status Code: 400; Error Code: MalformedXML; Request ID: AQ0N7SXGG4WVF5TE; S3 Extended Request ID: ikOb6kmhRrqIgZ/ql56OprE6zPvC2Q9cgbMJ/E4X6AOSZQW/JNalAvmcrTkZS/e5483twGIbE10=; Proxy: null)
	at com.google.exposurenotification.privateanalytics.ingestion.pipeline.IngestionPipelineIT.getTestObjKeys(IngestionPipelineIT.java:770)
	at com.google.exposurenotification.privateanalytics.ingestion.pipeline.IngestionPipelineIT.testIngestionPipelineAWS(IngestionPipelineIT.java:199)

[ERROR] com.google.exposurenotification.privateanalytics.ingestion.pipeline.IngestionPipelineIT.testIngestionPipeline  Time elapsed: 3.27 s  <<< FAILURE!
java.lang.AssertionError: No shares found in phaFolder
	at com.google.exposurenotification.privateanalytics.ingestion.pipeline.IngestionPipelineIT.checkSuccessfulFork(IngestionPipelineIT.java:604)
	at com.google.exposurenotification.privateanalytics.ingestion.pipeline.IngestionPipelineIT.testIngestionPipeline(IngestionPipelineIT.java:155)

[ERROR] com.google.exposurenotification.privateanalytics.ingestion.pipeline.IngestionPipelineIT.testIngestionPipelineDeviceAttestationEnabled  Time elapsed: 2.322 s  <<< FAILURE!
java.lang.AssertionError: expected:<1> but was:<0>
	at com.google.exposurenotification.privateanalytics.ingestion.pipeline.IngestionPipelineIT.testIngestionPipelineDeviceAttestationEnabled(IngestionPipelineIT.java:326)
```

I don't know how the S3 error got introduced, I don't see it when running the integration tests on `main`. I assume it happens since no data is written to S3 in the test. Are all these failures attributed to the above missing fix [apache/beam#15425 ](https://github.com/apache/beam/pull/15425)

**Comment by BenWhitehead** [comment]
Both of `IngestionPipelineIT.testIngestionPipeline` and `IngestionPipelineIT.testIngestionPipelineDeviceAttestationEnabled` are currently expected to fail until apache/beam#15425 is released in beam 2.34.0.

Regarding the S3 failure, I'm less certain. It's possible it's failing because data isn't loaded due to the same bug. I don't have access to S3 and can't debug this locally like we could for the other two. 

I'm going to push a commit where I try to add the apache snapshot repo into the build which has a 2.34.0-SNAPSHOT available which includes the fix.

**Review by hostirosti — APPROVED** [review]
LGTM

**Review by gurayAlsac — APPROVED** [review]

**Pull request merged**